packages/core/src/behavior.ts

```diff
@@ -67,59 +66,59 @@ export function createDeferredBehavior<TEvent extends EventObject>(
   let dispose;
 
   const behavior: Behavior<TEvent, undefined> = {
-    receive: (_, event) => {
-      const plainEvent = isSCXMLEvent(event) ? event.data : event;
-      receivers.forEach((receiver) => receiver(plainEvent));
-
-      return behavior;
-    },
-    receiveSignal: (actorContext, signal) => {
-      if (signal === startSignal) {
-        const sender: Sender<TEvent> = (e) => {
-          if (canceled) {
-            return;
-          }
+    receive: (_, event, actorContext) => {
+      if (isSignal(event)) {
```

Comment:
in the body of this block `event` is still being checked against concrete signals (start & stop) so this `isSignal` check seems somewhat redundant here

Reply:
We need to check if it's a signal; otherwise TypeScript complains later. This has been refactored.